app/lib/actions/recurring_action.rb

```diff
@@ -0,0 +1,62 @@
+module Actions
+  class RecurringAction < Actions::ActionWithSubPlans
+
+    include ::Dynflow::Action::Cancellable
+
+    middleware.use Actions::Middleware::KeepCurrentUser
+    middleware.use Actions::Middleware::InheritTaskGroups
+
+    def delay(delay_options, recurring_logic, action, *args)
+      task.add_missing_task_groups(recurring_logic.task_group)
+      action_class = action.is_a?(Class) ? action : action.constantize
+      action = action.to_s if action.is_a?(Class)
+      action_class.serializer_class.new [recurring_logic, action] + args
+    end
+
+    def plan(recurring_logic, action, *args)
+      serialized_args = serializer_class.new(args).tap { |s| s.perform_serialization! }.serialized_args
+      if recurring_logic.new_record?
+        plan_action(RecurringLogic::Create, recurring_logic)
+      else
+        task.add_missing_task_groups(recurring_logic.task_group)
+        plan_self :action => action.to_s,
+                  :serialized_args => serialized_args
+      end
+      trigger_repeat(recurring_logic, action, *args) if recurring_logic.can_continue?
+    end
+
+    def create_sub_plans
+      [input[:action].constantize].map { |action_class| trigger action_class, *deserialized_args }
```

Comment:
The `create_sub_plans` dosn't have to return just array: single trigger call is just fine, no need to do the `map` there